ClinGen/clincoded
# Connects to #915 kl bug brackets no disease

**This PR has changes for parsing and displaying ClinVar interpretation data in Basic Info tab, including:**
 1. Only retrieve and display "primary" RCVs.
 2. Retrieve and display not only "Disease" type but also all other types of condition associated to a RCV.
 3. Add external links to HPO and MeSH. In case source name of an identifier is not "MedGen", "Orphanet", "OMIM", "HPO" or "MeSH", display identifier as "source name: id" without link since pre-set urls to external web pages are limited to the 5 sources above currently.

**Test Steps**
* Test 1 (primary RCVs)
  - In VCI, use ClinVar variant "15333" and expect to see only 4 primary RCVs (RCV000016575, RCV000016573, RCV000016574 and RCV000224000) as:
![screen shot 2016-09-18 at 11 01 20 pm](https://cloud.githubusercontent.com/assets/9206696/18623617/49dddc9c-7df4-11e6-9cc3-64a6a26c4a53.png)
Related meta data in ClinVar xml file: 
![screen shot 2016-09-18 at 11 10 17 pm](https://cloud.githubusercontent.com/assets/9206696/18623698/59551fa4-7df5-11e6-9814-0f787c79b69e.png)
xml file at: [https://www.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=clinvar&rettype=variation&id=15333](https://www.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=clinvar&rettype=variation&id=15333)

* Test 2 (Non-disease type condition)
  - Click `New Variant Curation` button at top, enter variant id "60671", click `Save and View Evidence` and expect to see ClinVar interpretation data as:
![screen shot 2016-09-18 at 7 16 57 pm](https://cloud.githubusercontent.com/assets/9206696/18621187/805e6cde-7dd4-11e6-8ca8-db3279aada57.png)
Related ClinVar meta data `Type="DrugResponse"` as:
![screen shot 2016-09-18 at 11 23 47 pm](https://cloud.githubusercontent.com/assets/9206696/18623887/1d4b387a-7df7-11e6-8008-6b4297065453.png)
in xml file [https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=clinvar&rettype=clinvarset&id=RCV000054486](https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=clinvar&rettype=clinvarset&id=RCV000054486)

* Test 3 (Links to MeSH and HPO)
  - Click `New Variant Curation`, enter variant "13969",  click `Save and View Evidence` and expect to see identifier "MeSH" with a link:
![screen shot 2016-09-21 at 1 06 50 pm](https://cloud.githubusercontent.com/assets/9206696/18727170/594fa296-7ffc-11e6-9cc5-99fd7c1e2d45.png)
  - Click the link and expect to open a MeSH web page in a new browser tab as:
![screen shot 2016-09-21 at 1 02 41 pm](https://cloud.githubusercontent.com/assets/9206696/18727057/e4f8a154-7ffb-11e6-88ff-c21de8f7c246.png)
  - Click `New Variant Curation`, enter variant "156411",  click `Save and View Evidence` and expect to see "HPO" identifiers with links as:
![screen shot 2016-09-19 at 11 32 13 pm](https://cloud.githubusercontent.com/assets/9206696/18659734/52378cda-7ec1-11e6-91fc-4cae1bcefc4b.png)
  - Click a HPO link and expect to see HPO web page for relevant condition in a new browser tab.

**End of Test**
